src/Microsoft.AspNetCore.SignalR.Common/Protocol/HandshakeResponseMessage.cs

```diff
@@ -11,21 +11,36 @@ public class HandshakeResponseMessage : HubMessage
         /// <summary>
         /// An empty response message with no error.
         /// </summary>
-        public static readonly HandshakeResponseMessage Empty = new HandshakeResponseMessage(null);
+        public static readonly HandshakeResponseMessage Empty = new HandshakeResponseMessage(error: null);
 
         /// <summary>
         /// Gets the optional error message.
         /// </summary>
         public string Error { get; }
 
+        /// <summary>
+        /// Highest minor protocol version that the server supports.
+        /// </summary>
+        public int MinorVersion { get; }
+
         /// <summary>
         /// Initializes a new instance of the <see cref="HandshakeResponseMessage"/> class.
+        /// If you are intending to send back an error, the minor version doesn't matter.
         /// </summary>
-        /// <param name="error">An optional response error message. A <c>null</c> error message indicates a succesful handshake.</param>
-        public HandshakeResponseMessage(string error)
+        /// <param name="error"></param>
+        public HandshakeResponseMessage(string error) : this(null, error) { }
+
+        public HandshakeResponseMessage(int minorVersion) : this(minorVersion, null) { }
```

Comment:
Add some documentation comments to this as well. Including a `<param>` tag for the `minorVersion` that explains it's purpose.